test/index.spec.ts

```diff
@@ -174,6 +174,97 @@ describe('Avro mock data generator', () => {
     });
   });
 
+  it('lets the caller pick union by short name', () => {
+    // 1/1000 chances that this test falsely pass is deemed acceptable, but you can always up the number
+    const unionType = Array(1000).fill({
+      type: 'record',
+      name: 'CityFarm',
+      fields: [{ name: 'nbPidgeons', type: 'int' }],
+    });
+    unionType[550] = {
+      type: 'record',
+      name: 'CountryFarm',
+      fields: [{ name: 'nbChickens', type: 'int' }],
+    };
+
+    const result = generateData(
+      {
+        type: 'record',
+        fields: [
+          {
+            name: 'farm',
+            type: unionType,
+          },
+        ],
+      },
+      { pickUnion: ['CountryFarm'] },
+    );
+    expect(result).toEqual({
+      farm: { CountryFarm: { nbChickens: expect.any(Number) } },
+    });
+  });
+
+  it('lets the caller pick union by full name', () => {
+    // 1/1000 chances that this test falsely pass is deemed acceptable, but you can always up the number
+    const unionType = Array(1000).fill({
+      type: 'record',
+      name: 'CityFarm',
+      fields: [{ name: 'nbPidgeons', type: 'int' }],
+    });
+    unionType[550] = {
+      type: 'record',
+      name: 'CountryFarm',
+      fields: [{ name: 'nbChickens', type: 'int' }],
+    };
+
+    const result = generateData(
+      {
+        type: 'record',
+        namespace: 'my.lovely',
+        fields: [
+          {
+            name: 'farm',
+            type: unionType,
+          },
+        ],
+      },
+      { pickUnion: ['my.lovely.CountryFarm'] },
+    );
+    expect(result).toEqual({
+      farm: { 'my.lovely.CountryFarm': { nbChickens: expect.any(Number) } },
+    });
+  });
+
+  it('supports top level union types', () => {
+    const result = generateData([
+      {
+        type: 'record',
+        name: 'Owners',
+        namespace: 'com.farms',
+        fields: [
+          {
+            name: 'name',
+            type: 'string',
+          },
+        ],
+      },
+      {
+        type: 'record2',
```

Comment:
Should this not be of type `record` and then add `'com.farms.Animals'` to the expectation?

Reply:
oh you have a good point re `record2`. I don't know what happened in my brain.

As it's a union type , only one of the 2 types will be picked (the first one by default). The expectation not including `Animals` is expected, if it had contained it then the wrong type would have been chosen.